benchmark/run_benchmark.py

```diff
@@ -90,7 +90,7 @@ async def run_userspace_uprobe_test():
     await server_start_cb.wait()
     print("server started!")
     result = None
-    for i in range(100):
+    for i in range(10):
```

Comment:
Wouldn't reducing number of iterations reduce the reliability of our results?

Reply:
No. it will not bring too much difference.

For example, the uprobe time in my machine is about 1000ns (kernel 10000ns)

When we have a true value of 1000, with each measurement having an error (standard deviation) of approximately 100, the measurement results can be assumed to follow a normal distribution \( X \sim N(1000, 100^2) \).

### Standard Error of the Mean
The standard error (SE) of the sample mean decreases as the sample size increases. The formula for SE is:

\[
\text{SE} = \frac{\sigma}{\sqrt{n}}
\]

where:
- \( \sigma = 100 \) (the standard deviation of measurement errors)
- \( n \) is the sample size.

- For a sample size of 10, the SE is approximately 31.62.
- For a sample size of 100, the SE reduces to 10.

### Error Comparison

- **With 10 samples**: The standard error of the mean is 31.62. This means the sample mean is expected to be within approximately ±31.62 of the true value (1000).
- **With 100 samples**: The standard error of the mean is 10, meaning the sample mean is expected to be within ±10 of the true value.

### Conclusion
While increasing the sample size from 10 to 100 does reduce the standard error, the reduction is not drastically large. The standard error decreases from 31.62 to 10, which improves the precision of the sample mean but does not imply a massive difference in accuracy. Thus, the error in estimating the true value with 10 measurements versus 100 measurements is somewhat significant but not overwhelmingly so. In practical terms, even with 10 measurements, the sample mean is expected to be reasonably close to the true value.

Reply:
Thank you so much for this detailed explanation. Got it.